utilities/utils/response.go

```diff
@@ -98,6 +98,9 @@ func getInstanceHostMapData(event *revents.Event, client *client.Client, cache *
 	if err != nil {
 		return map[string]interface{}{}, errors.Wrap(err, constants.GetInstanceHostMapDataError+"failed to get mount data")
 	}
+	if err := setupDNS(inspect.ID); err != nil {
```

Comment:
This is not the correct place to make this call. It would probably be more appropriate at the end of this function:
https://github.com/rancher/agent/blob/master/core/compute/compute.go#L22